### DISCUSSION TITLE
With tycho, when the dependencies are optional, we are getting compilation error while building.

### DISCUSSION THREAD

**Discussion by skumarjh-tibco** [eclipse-tycho/tycho#4291] [General]
Hello experts,
I have a maven project where I am using some optional dependencies in manifest file.
e.g. com.test.ftl;version="[4.0.0,7.0.0)";resolution:=optional,
 com.test.ftl.exceptions;version="[4.0.0,7.0.0)";resolution:=optional
And we do import and consume the methods from these packages in the classes.

The dependencies are optional in nature.
With tycho, when the dependencies are optional, we are getting compilation error while building.

Can someone plesae help me on this ?

Thanks!

**Comment by laeubi** [comment]
I think you need to provide an example. In general marking things as optional is quite tricky to get correct and one should avoid this.

**Comment by skumarjh-tibco** [comment]
[OptionalDependencyIssue.log](https://github.com/user-attachments/files/17071215/OptionalDependencyIssue.log)

Hi @laeubi , thank you for your response. 
As you mentioned, I have attached the related example for your reference. 
Please find the attachments where it includes the Manifest file where the dependencies are optional, the class where I am using it, the error that I am getting while building the plugin and the pom.xml file. 

Please help me to resolve it if I am missing something.

Thanks!

**Reply by laeubi** [reply]
```
<dependency-resolution>
          <optionalDependencies>ignore</optionalDependencies>
 </dependency-resolution>
```

This looks wrong as you effectively tell Tycho to not resolve optional dependencies.

**Reply by skumarjh-tibco** [reply]
Thanks a lot for your response.
Found this after posting the reply.
Removed it and everything worked well.

I will close the discussion now.
Thank you again.

**Comment by skumarjh-tibco** [comment]
Got the resolution. 
Thanks!
